### ISSUE TITLE
Expand Development Documentation

### ISSUE THREAD

**Issue by RPG-Alex** [buildsrs/buildsrs#4]
Perhaps the development documentation can be expanded with more info

**Comment by xfbs** [comment]
Definitely! I had a chat with @0xpr03 yesterday and we also agreed that we need to do a better job of documenting things. 

I have just merged a PR that includes a development guide (it takes me a surprisingly long amount of time to write this out in a way that is sensible). That guide is [here](https://github.com/buildsrs/buildsrs/blob/main/docs/guides/development.md). It is somewhat comprehensive, currently lacks information on how to launch the frontend (which currently is just a placeholder anyways).

But going further than that, the goal is to document (and collaborate on) architectural decisions using a RFC-like process. I will be traveling this weekend and ideally use that time to document the decisions we have made so far such as to get you up to speed (and so we all have some common reference). 

Besides doing a better job of documenting things, we have also agreed that the short-term goal is to get something alpha-level ready and deployed. At current speed, I would assume that this is something we could accomplish in the next week or two, and then we can take it from there.
